pkg/distributor/validate.go

```diff
@@ -73,17 +74,12 @@ var (
 		"received a series whose number of labels exceeds the limit (actual: %d, limit: %d) series: '%.200s%s'",
 		validation.MaxLabelNamesPerSeriesFlag,
 	)
-	noMetricNameMsgFormat              = globalerror.MissingMetricName.Message("received series has no metric name")
-	invalidMetricNameMsgFormat         = globalerror.InvalidMetricName.Message("received a series with invalid metric name: '%.200s'")
-	maxNativeHistogramBucketsMsgFormat = fmt.Sprintf(
-		"received a native histogram sample with too many buckets, timestamp: %%d series: %%s, buckets: %%d, limit: %%d (%s)",
-		globalerror.MaxNativeHistogramBuckets,
-	)
-	notReducibleNativeHistogramMsgFormat = fmt.Sprintf(
-		"received a native histogram sample with too many buckets and cannot reduce, timestamp: %%d series: %%s, buckets: %%d, limit: %%d (%s)",
-		globalerror.NotReducibleNativeHistogram,
-	)
-	sampleTimestampTooNewMsgFormat = globalerror.SampleTooFarInFuture.MessageWithPerTenantLimitConfig(
+	noMetricNameMsgFormat                 = globalerror.MissingMetricName.Message("received series has no metric name")
+	invalidMetricNameMsgFormat            = globalerror.InvalidMetricName.Message("received a series with invalid metric name: '%.200s'")
+	maxNativeHistogramBucketsMsgFormat    = globalerror.MaxNativeHistogramBuckets.Message("received a native histogram sample with too many buckets, timestamp: %d series: %s, buckets: %d, limit: %d")
+	notReducibleNativeHistogramMsgFormat  = globalerror.NotReducibleNativeHistogram.Message("received a native histogram sample with too many buckets and cannot reduce, timestamp: %d series: %s, buckets: %d, limit: %d")
+	invalidSchemaNativeHistogramMsgFormat = globalerror.InvalidSchemaNativeHistogram.Message("received a native histogram sample with an invalid schema %d: valid schema numbers are %d <= n <= %d")
```

Comment:
Beorn is suggesting that we use some high schema number (e.g. 127) for custom histograms, so this would have to be rewritten later and any time we add a new schema. I think the documentation in the runbook is enough, also there we can link to the mimir docs for more details.
```suggestion
	invalidSchemaNativeHistogramMsgFormat = globalerror.InvalidSchemaNativeHistogram.Message("received a native histogram sample with an invalid schema %d")
```

Reply:
I have applied the requested change.